lib/__tests__/cloudworker-e2e.test.js

```diff
@@ -211,4 +211,186 @@ describe('cloudworker-e2e', async () => {
+        return s;
+      }
+
+      async function respondWithHelloAsSHA256() {
+        const sha = await toSHA256("hello")
+        return new Response(sha, {status: 200})
+      }
+
+      async function toSHA256(message) {
+        let hash = await crypto.subtle.digest('SHA-256', byteStringToUint8Array(message))
+        return toHexString(hash)
+      }
+    `
+    const server = new Cloudworker(script).listen(8080)
+    const res = await axios.get('http://localhost:8080', defaultAxiosOpts)
+    expect(res.status).toEqual(200)
+    // taken from `echo -n "hello" | shasum -a 256`
+    expect(res.data).toContain('2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824')
+    await server.close()
+    cb()
+  })
+
+  test('simple hmac sign', async (cb) => {
+    const script = `
+      addEventListener('fetch', event => {
+        const url = new URL(event.request.url)
+        event.respondWith(generateSignedUrl(url))
+      })
+
+      function toHexString(buffer) {
+        var s = '', h = '0123456789abcdef';
+        (new Uint8Array(buffer)).forEach((v) => { s += h[v >> 4] + h[v & 15]; })
+        return s;
+      }
+
+      async function generateSignedUrl(url) {
+        // We'll need some super-secret data to use as a symmetric key.
+        const secretKeyData = byteStringToUint8Array("my secret symmetric key")
+        const key = await crypto.subtle.importKey(
+          "raw", secretKeyData,
+          { name: "HMAC", hash: "SHA-256" },
+          false, [ "sign" ]
+        )
+
+        // Signed requests expire after one minute. Note that you could choose
+        // expiration durations dynamically, depending on, e.g. the path or a query
+        // parameter.
+        const expirationMs = 60000
+        const expiry = new Date('Feb 28 2013 19:00:00 UTC').getTime()
+        const dataToAuthenticate = '/verify/bob1362078000000'
+
+        const mac = await crypto.subtle.sign(
+          "HMAC", key,
+          byteStringToUint8Array(dataToAuthenticate)
+        )
+
+        // mac is an ArrayBuffer, so make it hex
+        const hex64Mac = toHexString(mac)
+
+        url.searchParams.set("mac", hex64Mac)
+        url.searchParams.set("expiry", expiry)
+
+        return new Response(url)
+      }
+
+      function byteStringToUint8Array(byteString) {
+        const ui = new Uint8Array(byteString.length)
+        for (let i = 0; i < byteString.length; ++i) {
+          ui[i] = byteString.charCodeAt(i)
+        }
+        return ui
+      }
+    `
+    const server = new Cloudworker(script).listen(8080)
+    const res = await axios.get('http://localhost:8080', defaultAxiosOpts)
+    expect(res.status).toEqual(200)
+    expect(res.data).toContain('?mac=ae0342a9b989d76cfc457439f615ed72651ba7bce1994361621a003f6ade5508&expiry=1362078000000')
+    await server.close()
+    cb()
+  })
+
+  test('simple hmac verify', async (cb) => {
+    const script = `
+      addEventListener('fetch', event => {
```

Comment:
For long e2e scripts like this, please add as a fixture file instead. It helps with readability.